java/datastructures/trees/lib/src/main/java/trees/BinaryTree.java

```diff
@@ -0,0 +1,97 @@
+package trees;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BinaryTree {
+  private Node root;
+
+  public BinaryTree() {
+    root = null;
+  }
+
+  public BinaryTree(Node root) {
+    this.root = root;
+  }
+
+  public void insert(int value) {
+    if (root == null) {
+      root = new Node(value);
+    } else {
+      insertHelper(root, value);
+    }
+  }
+
+  private void insertHelper(Node current, int value) {
+    if (value < current.getValue()) {
+      if (current.getLeft() == null) {
+        current.setLeft(new Node(value));
+      } else {
+        insertHelper(current.getLeft(), value);
+      }
+    } else {
+      if (current.getRight() == null) {
+        current.setRight(new Node(value));
+      } else {
+        insertHelper(current.getRight(), value);
+      }
+    }
+  }
+
+  public int[] preOrder(List<Integer> traversal) {
```

Comment:
Here, you're passing in a List called `traversal`, but then creating a new list called `result`. The method then adds all of the traversal values to `result`, so `traversal` is never changed. See also your inOrder and postOrder methods. Once I fixed this issue, all your tests passed!